### ISSUE TITLE
Explanation of the state of uBlock Origin (and other blockers) for Safari

### ISSUE THREAD

**Issue by ghost** [el1t/uBlock-Safari#158]
Very quick tl;dr: uBO will no longer work with Safari, use Firefox or a new "content blocker" app (see below for good replacements).

In the past few months, and especially in the past week, there have been a lot of posts and comments questioning the status of [uBlock Origin for Safari](https://github.com/el1t/uBlock-Safari). This should answer all questions on the status of uBO for safari.

uBlock Origin was ported for Safari in 2016, and was updated regulary (mostly changes from the main project) until 2018 when development completley stopped. Since then Apple has begun phasing out Safari extensions as extensions, and has instead been implenting a new extensions framework which is extremley limited in adblocking functions, only allowing "content blockers", which are just links bundled as an app which Safari enforces. From Safari 12 / macOS Mojave, old legacy Safari extensions were still allowed, but came with warnings saying that they will slow down your browsing (they infact won't, or at least not noticably). Safari also recently shut their Extension Gallery, instead redirecting it to the mac app store. Though it is still curently possible to install uBlock Origin [by downloading the extension from Github](https://github.com/el1t/uBlock-Safari/releases/download/1.16.0/uBlock0.safariextz) (edit: must follow [these instructions](http://osxdaily.com/2019/02/08/bypass-safari-no-longer-supports-unsafe-extension-error-mac/), it will not be starting from Safari 13 / macOS Catalina, when the legacy entension API will be fully deprecated.

**It will not possible for uBlock Origin to work with the upcoming Safari 13 / macOS Catalina release**
If you are a current user of uBlock Origin for Safari here are the options to continue blocking ads:

1. For the moment continue to use Safari 12 with uBlockOrigin. Anybody with uBO currently installed, it won't be removed until you update to Safari 13. If you don't have uBO installed, and wish to install on a pre-Catalina version of Safari, [Download the latest (and final) release here](https://github.com/el1t/uBlock-Safari/releases/download/1.16.0/uBlock0.safariextz)  and follow [these instructions](http://osxdaily.com/2019/02/08/bypass-safari-no-longer-supports-unsafe-extension-error-mac/) to install it. Unfortunately it's a bit complicated. This will stop working with macOS Catalina (coming "this fall"). **Update:** It appears that it is not possible to install uBO permanently, it will always uninstall on a restart of Safari. If you have it, it *should* stay.
2. Switch to a different browser. If you choose this, I strongly recomend Firefox. Chrome will itself be ending support for uBlockOrigin soon. If battery life is an issue for you get [Firefox Beta, Nightly or Developer](https://www.mozilla.org/en-US/firefox/channel/desktop/) which has [massive battery life improvements to bring it on par with Safari / Chrome](https://appleinsider.com/articles/19/09/03/firefox-macos-test-build-is-far-better-for-your-battery-life) being tested (note: somewhat unstable). This will come to the stable version, hopefully in time for uBO-Safari's eol.
3. Get a content blocker. Not nearly as powerful as uBO, but the best option if you want to stay with Safari. **Do not get the app called "uBlock"**, this is unassociated with uBlockOrigin ([read about the split here](https://en.wikipedia.org/wiki/UBlock_Origin#History)), and is simply a content blocker with a big negative feature of having acceptable ads built in (which is AdBlockPlus's pay-to-play ad and tracker unblocking program). It shares no code with uBO and has no advantages over any other content blocking app. Here are some recomendations of content blockers:

**Top picks**

- [Adguard for Safari](https://apps.apple.com/app/adguard-for-safari/id1440147259), very good blocking and free. Very customisable. Doesnt always have the best performance (though this has apparently been fixed). AdGuard's performance is quite controversial, read discussion [here].(https://github.com/el1t/uBlock-Safari/issues/158#issuecomment-533546184). AdGuard also uses the same lists and filter syntax as uBO, making migration easier.
~~- [1Blocker](https://apps.apple.com/us/app/1blocker-faster-safer-web/id1107421413), by far the most powerful, with custom rules and an element picker. This is the best option if you don't mind paying. Costs $10. ~~**Update:** do not buy. [1Blocker has teased 1BlockerX for Mac, which will be a paid upgrade. I cannot reccomend buying an app now which will soon be obsolete.](https://mobile.twitter.com/1BlockerApp).~~ **Update 2:** 1Blocker has said that it will be a free upgrade to 1BlockerX from the current version, so there isn't any reason to wait if you want it.~~ 1Blocker has switched to a subscription model, even for people who already paid for it. Avoid.
- [Ka-block](https://apps.apple.com/us/app/ka-block/id1335413823). Unexceptional. Free. Will block ads, nothing more. Pretty sure it's just a wrapper for easylist, but it's a good option if you mind paying for 1Blocker. Also open-source.

**Other Good Options**

- [Ghostery Lite](https://apps.apple.com/us/app/ghostery-lite/id1436953057). Free. Ghostery. Some advanced options for whitelisting. Good lists for ad and tracker blocking.
- [Adguard for Mac](https://adguard.com/en/adguard-mac/overview.html). Fully featured system wide adblocker, contains custom lists and element picker. Does cost after a trial, [see here for prices](https://adguard.com/en/license.html).
- [Wipr](https://apps.apple.com/us/app/wipr/id1320666476). $1.99, simple featureless and popular. ~~Don't see any advantage in this over Ka-block (see above) for an extra $1.99.~~ Apparently Ka-Block doesn't work for youtube (wipr does), and Wipr uses 3 extensions to get around the limit in rules.

**Do Not Reccomend**

- AdBlock Plus for Safari - Supports acceptable ads, a pay-to-play ad allowing system which allows certain ads and trackers which meet guidelines and pay AdBlock Plus. Some of these ads, imo, are not acceptable, and I don't consider any trackers acceptable. Uses Easylist so otherwise is identical to Ka-Block!.
- uBlock - Don't at all associated with uBO or the code which uBO contains. Is instead identical to AdBlock Plus in all but name including acceptable ads.
- AdBlock for Safari (made by BETAFISH INC) - Yet another acceptable ads-supporting blocker which just uses easylist. Avoid.
- There are plenty more on the mac app store, have a look if none of these suit. No new content blockers can spy on you as they send lists though Safari's built in system, so they are all pretty safe. If you find a good one comment and I'll add it to this list.

**Update:** [Here is a statement from gorhill (uBO developer) on the state of Safari](https://github.com/gorhill/uBlock/blob/master/platform/safari/README.md#safari-platform)

**Edit:** a lot people are asking about uBlock Origin not working in the future on Chrome. If you'd like more information on this, [here is an article from ghacks from january](https://www.ghacks.net/2019/01/22/chrome-extension-manifest-v3-could-end-ublock-origin-for-chrome/), and a [statement from gorhill, developer of uBlock](https://bugs.chromium.org/p/chromium/issues/detail?id=896897&desc=2#c23).**

There has been discussion of this on [Reddit](https://old.reddit.com/r/uBlockOrigin/comments/d18cpi/explanation_of_the_state_of_ublock_origin_and/) [Github](https://github.com/el1t/uBlock-Safari/issues/158) and [Hacker News](https://news.ycombinator.com/item?id=21025252).

**Comment by dpkonofa** [comment]
What differentiates an "ad-blocker" from a "content-blocker" that makes it impossible for this change? Isn't an "ad-blocker" just a "content-blocker" that only blocks a very specific type of content?

**Comment by Tongzhao9417** [comment]
thanks for u summation. it's so sad:( i have used it for three years and it's time to say goodbye...

**Comment by ghost** [comment]
> What differentiates an "ad-blocker" from a "content-blocker" that makes it impossible for this change? Isn't an "ad-blocker" just a "content-blocker" that only blocks a very specific type of content?

I must admit the the terminology isn't very clear. A Safari "content-blocker" app sends a list to Safari, and Safari blocks it. A regular blocker (like uBO) blocks content itself. Safari content blockers aren't all bad, they are more secure in that they can't possibly collect your browsing history (not that uBO does), but lack the level of customisation and power that a regular blocker like uBO can provide.

**Comment by nheeren** [comment]
Thanks! You should [pin](https://help.github.com/en/articles/pinning-an-issue-to-your-repository) this issue.

**Comment by ghost** [comment]
> Thanks! You should [pin](https://help.github.com/en/articles/pinning-an-issue-to-your-repository) this issue.

only @el1t can do that

**Comment by Rjevski** [comment]
Just FIY I would not recommend buying 1Blocker. I paid both for the initial version and the "1Blocker X" one on IOS but I had the constant feeling the filter lists weren't being updated and I wasn't sure of their origin (I'm assuming they use EasyList for the ads, but which one do they use for privacy & annoyances?).

AdGuard for Safari is free, open-source and is compatible with uBlock-style filter lists (it has code that does its best to convert them to Safari content blocking lists). You can choose to use EasyList, EasyPrivacy, Fanboy's lists or any of the regional lists so I would recommend it over 1Blocker.

**Comment by ameshkov** [comment]
> Adguard for Safari, the most effective, but contains electron (chromium). I'm not entirely sure how it works, but it will cost performance and battery life on your computer so I would avoid it.

1. Electron is used for the UI part of it only. All other parts are native.
2. Running the electron part in the background is not necessary. Although, it is recommended so that AG could update filters automatically.
3. The linked issue was [resolved](https://github.com/AdguardTeam/AdGuardForSafari/issues/210) in the latest release

> I'm not entirely sure how it works

You can take a look - https://github.com/AdguardTeam/AdGuardForSafari

**Comment by ghost** [comment]
> > Adguard for Safari, the most effective, but contains electron (chromium). I'm not entirely sure how it works, but it will cost performance and battery life on your computer so I would avoid it.
> 
>     1. Electron is used for the UI part of it only. All other parts are native.
> 
>     2. Running the electron part in the background is not necessary. Although, it is recommended so that AG could update filters automatically.
> 
>     3. The linked issue was [resolved](https://github.com/AdguardTeam/AdGuardForSafari/issues/210) in the latest release
> 
> 
> > I'm not entirely sure how it works
> 
> You can take a look - https://github.com/AdguardTeam/AdGuardForSafari

Thank you for responding. Ive updated the post and corrected my errors. 

Based on point 2, for filters to update, the electron part needs to be running. Does leaving this running for filters to update have any serious power or performance issues?

**Comment by ghost** [comment]
> Just FIY I would not recommend buying 1Blocker. I paid both for the initial version and the "1Blocker X" one on IOS but I had the constant feeling the filter lists weren't being updated and I wasn't sure of their origin (I'm assuming they use EasyList for the ads, but which one do they use for privacy & annoyances?).
> 
> AdGuard for Safari is free, open-source and is compatible with uBlock-style filter lists (it has code that does its best to convert them to Safari content blocking lists). You can choose to use EasyList, EasyPrivacy, Fanboy's lists or any of the regional lists so I would recommend it over 1Blocker.

thats odd, when I wrote this post, everything I could find had it as the best or one of the best. [This guy](https://brooksreview.net/2018/09/safari-content-blocker-evaluations-92618-edition/) tested them quite thoroughly from 2015 to 2018, and found 1Blocker to stand out. Looking at it again now, it seems that 1Blocker requires an app update to update their lists (entirely based on the changelog), which would explain the slow rate of updates. Tommorow I will investigate further and update my list once again.

**Comment by ameshkov** [comment]
> Based on point 2, for filters to update, the electron part needs to be running. Does leaving this running for filters to update have any serious power or performance issues?

Not really. Here's how it looks on my MBP, and in my case, I am actually actively using it, not just keep it in the background:
![](https://uploads.adguard.com/up04_9xjg6_Activity_Monitor_Applications_in_last_12_hours.png)

There was indeed an issue with the version of Electron we were using previously, but it's now gone. 

On the other hand, I share your sentiments about Electron. The UI is generally worse and slower than it could be if we made it native. But it allowed reusing a lot of existing code from the Chrome/FF extensions so here we are.

**Comment by freediverx** [comment]
AdGuard is SLOOOOOOOOW. Horrible piece of software. Typical of garbage created with Electron.

**Comment by freediverx** [comment]
For anyone claiming AdGuard is “fine”…


When you first click on the AdGuard toolbar button you get this:





Click on the Run button and you have to wait 8 seconds for this monstrosity to load:



Close that eyesore and go back to Safari. Clock on the AdGuard button again and it always takes a full 2 seconds for the following pop-up to appear:




And of course then you have to go through a multi-step process for selecting a screen element on the page, experimenting with the slider, previewing, and then saving your change. Nothing responds to keyboard commands like hitting the Escape button because of course it doesn’t.

**Comment by benjamingr** [comment]
What about building uBlockOrigin as either a proxy with a root certificate or using applescript and making it a Mac app?

**Comment by vassudanagunta** [comment]
Further options are being discussed in uBlockOrigin/uBlock-issues#145.

**Comment by vassudanagunta** [comment]
@p4t44, you may want to link to the HackerNews discussion of the whys and pros and cons of Apple's change to Safari: https://news.ycombinator.com/item?id=21025252

**Comment by ghost** [comment]
> > Based on point 2, for filters to update, the electron part needs to be running. Does leaving this running for filters to update have any serious power or performance issues?
> 
> Not really. Here's how it looks on my MBP, and in my case, I am actually actively using it, not just keep it in the background:
> ![](https://camo.githubusercontent.com/40eb13255531276d09bdb69bc36d03e04920328e/68747470733a2f2f75706c6f6164732e616467756172642e636f6d2f757030345f39786a67365f41637469766974795f4d6f6e69746f725f4170706c69636174696f6e735f696e5f6c6173745f31325f686f7572732e706e67)
> 
> There was indeed an issue with the version of Electron we were using previously, but it's now gone.
> 
> On the other hand, I share your sentiments about Electron. The UI is generally worse and slower than it could be if we made it native. But it allowed reusing a lot of existing code from the Chrome/FF extensions so here we are.

As AdGuard's performance is far from settled, I've linked here. If it goes fast for you, use it. If it goes slow for you, don't use and delete it.

**Comment by xtian** [comment]
Just FYI for anyone reading this thread: AdGuard for Mac requires installing a kernel extension which as far as I know none of the other options do. I didn't feel comfortable with that so I went with one of the other options.

**Comment by El-Fitz** [comment]
I would also suggest Little Snitch or some free equivalent, using block lists. Since it's a firewall, it won't be impacted by any changes in Safari.

**Comment by CrazyPython** [comment]
I was using a manually installed build of uBlock Origin on Safari, but Safari crashed twice, and uBlock was dissabled.

**Comment by kfur** [comment]
@p4t44 @freediverx Do you need nifty macos app or just good ad blocker ? 
You don't need to run it. Electron app i needed only for rule updates. It can't slow down your safari browser. So i think slowness of this app it's not critical issue. You can run it from time to time just for update rules.

**Comment by kfur** [comment]
@El-Fitz but functionality of this ad blocker will be limited to blocking hosts when real ad blockers much more powerful and it will diffidently slow down your browser. Better use dnscrypt-proxy with blocklist in this case if you could satisfied by host blockers.

**Comment by ameshkov** [comment]
@p4t44 
> As AdGuard's performance is far from settled, I've linked here. If it goes fast for you, use it. If it goes slow for you, don't use and delete it.

Why? It seems settled that the UI is slow as it's Electron-based, and I don't argue with that. It just has nothing to do with ad blocking performance, and there's no one on this thread complaining about that. Also, "slow" does not mean that it consumes additional energy or whatever. It simply takes more time to render when you decide to use it.

**Comment by cosmarc** [comment]
Just wanted to leave this here: https://thehackernews.com/2019/09/browser-chrome-extension-adblock.html

**Comment by cosmarc** [comment]
Just wanted to leave this here: https://thehackernews.com/2019/09/browser-chrome-extension-adblock.html

**Comment by ghost** [comment]
> @p4t44
> 
> > As AdGuard's performance is far from settled, I've linked here. If it goes fast for you, use it. If it goes slow for you, don't use and delete it.
> 
> Why? It seems settled that the UI is slow as it's Electron-based, and I don't argue with that. It just has nothing to do with ad blocking performance, and there's no one on this thread complaining about that. Also, "slow" does not mean that it consumes additional energy or whatever. It simply takes more time to render when you decide to use it.

It used to be that it was slower even without the UI open before a recent update. It is also reccomended that you leave the app open constantly for filters to update (which or may not slow it down). I can link you to a dozen reports of AdGuard going slow, so I think it is unreasonable not to mention it in the list.

Slow means that 1) it takes longer to use, 2) uses excessive CPU and memory to make the rest of your computer go slower and 3) use more battery. I'm really not sure wether it is significantly slower then alternatives, and I don't have the time or will to figure it out. I've linked to this thread so anybody unsure can read the comments of the AdGuard developer and decide for himself.

I think that if you are considering AdGuard, download it. Check activity monitor, if it takes up a lot of CPU or battery, uninstall and get something else. If it doesn't, then don't and use it.

**Comment by Sangeppato** [comment]
I would just like to mention a couple of advantages that Wipr has over Ka-Block:

* Wipr works with YouTube, while Ka-Block doesn't
* Wipr is actually made of 3 extensions in order to be able to overcome the 50k rules limitation imposed by Apple, so it can go up to 150k and block more stuff (Ka-Block can fail to block ads on some websites in my experience)
* I don't know if Ka-Block can do this, but Wipr can automatically update the filter list in background without having the app up and running all the time.

Note that with the new Content Block API you don't need to have a white-list option built into the extension, because you can exclude content blocking on specific websites directly from Safari going to Preferences -> Websites -> Content Blockers

**Comment by ghost** [comment]
> I would just like to mention a couple of advantages that Wipr has over Ka-Block:
> 
>     * Wipr works with YouTube, while Ka-Block doesn't
> 
>     * Wipr is actually made of 3 extensions in order to be able to overcome the 50k rules limitation imposed by Apple, so it can go up to 150k and block more stuff (Ka-Block can fail to block ads on some websites in my experience)
> 
>     * I don't know if Ka-Block can do this, but Wipr can automatically update the filter list in background without having the app up and running all the time.
> 
> 
> Note that with the new Content Block API you don't need to have a white-list option built into the extension, because you can exclude content blocking on specific websites directly from Safari going to Preferences -> Websites -> Content Blockers

Updated the post.

**Comment by tcelestino** [comment]
If I've known about this, I wouldn't update Safari 😢

**Comment by El-Fitz** [comment]
@kfur true; it doesn't offer the same level of ad blocking. But how would it slow down the browser? It doesn't run in it, and the requests just fail before even leaving the computer

**Comment by anywayaura** [comment]
bois ghostery lite is pretty good

**Comment by kfur** [comment]
@El-Fitz i think LS was developed not for it. I assume that software like dnscrypt-proxy or system host file will work faster.

**Comment by speckz** [comment]
FYI, you can install Firefox Focus on iOS and use that as a content blocker in Safari. See https://support.mozilla.org/en-US/kb/focus#w_step-one-allow-safari-to-use-content-blockers-you-only-need-to-do-this-once

**Comment by flinhong** [comment]
I've changed to Firefox now in all my platforms.

And testing the AdGuardHome on a server and then use the DNS filter through Argo tunnel on my Mac for Safari. Seems that this only works as a regular content blocker.

**Comment by greener2015** [comment]
I have downloaded MS Edge Chromium base. UBlock Origin works with it.

**Comment by superlex** [comment]
Hello :)

@p4t44 
> Safari content blockers aren't all bad, they are more secure in that they can't possibly collect your browsing history (not that uBO does), but lack the level of customisation and power that a regular blocker like uBO can provide.

Yep. Except for AdGuard Icon and AdGuard Advanced Blocking, all AdGuard extensions don't have access to the page, so they are more secure than uBlock Origin, for example. Then, because they use Safari content blocking API, they can block the elements before they're loaded.

---

I think AdGuard for Safari is the best option if you like uBlock Origin: 
1) uBlock0 and AdGuard share almost the same filter lists (EasyList, AdGuard Base, EasyPrivacy, Adguard Tracking Protection, etc..), so you have the same - best - ad blocking quality ;)
2) uBlock0 and AdGuard share almost the same filter syntax, so you can use your user filter list with both of them;
2.1) AdGuard introduced scriptlets, that are almost compatible with uBlock0 js (see compatibility table [here](https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/compatibility-table.md#scriptlets));
2.2) Compatibility with uBlock0 filters will be improved in the future (see issue [here](https://github.com/AdguardTeam/AdguardBrowserExtension/issues/1452)), so you can use uBlock0 filter lists as custom ones.

In addition:
- with AdGuard for Safari you can inject custom javascripts, for example in order to write a cookie, automatically click an element, etc..
- you can use uBlock Origin with Firefox and Chrome, basically; on the contrary, you can use AdGuard (so share all you custom rules, filters lists, etc) with all browsers and OS (Windows, macOS, Android, iOS.. ), and this is a great added value :)

PS: 
I don't like electron, but as @ameshkov said:
> Running the electron part in the background is not necessary

open it - update filters - close it.

PPS:
Thanks @p4t44 for this - explanation of the state of uBlock Origin for Safari - issue ;)

**Comment by ghost** [comment]
could ub0 be installed as an app, just like [iina](https://github.com/iina/iina) do with it's safari extension?

**Comment by alvarnell** [comment]
I’m sure it could, but  as explained in the initial posting here, it would not be able to provide much of the functionality that made it the best available. So it would be a wasted effort.

Sent from my iPad

-Al-

**Comment by ghost** [comment]
> could ub0 be installed as an app, just like [iina](https://github.com/iina/iina) do with it's safari extension?

An app could be made with similar functionality to uBlock Origin, but it wouldn't be uBlock Origin, as it would work completley differently.

**Comment by troymaclure** [comment]
**Adguard for MacOS** has the most complete list and is not restricted to the 50000 rules like the Safari extension. You can even add custom lists from [Filterlist](https://filterlists.com) for a more fine tuned list like Facebook ads or Youtube comments and so on. The price of the app is not cheap but it works system wide and as so on all browsers installed.

**Comment by superlex** [comment]
@stephanepiriou 
> Adguard for MacOS has the most complete list and is not restricted to the 50000 rules like the Safari extension.

Just to clarify: with AdGuard for Safari the total number of rules that can be used simultaneously is **300.000** (50.000 per group) :)

Btw, as you said, AdGuard for Mac can
> - filter your traffic in all browsers and apps on your Mac
> - have an unlimited number of filter rules
> - provide a better filtering quality (due to the lack of browser API restrictions)

(see details [here](https://github.com/AdguardTeam/AdGuardForSafari) and [here](https://github.com/AdguardTeam/AdguardForMac))

**Comment by isdn** [comment]
FYI, Adguard for Mac, as a system-wide app, installs its own root SSL certificate to the system and does MitM. And it is not open source.

**Comment by nidlol** [comment]
best option for Safari users? to stay on Safari 12.1.2 😬

**Comment by ghost** [comment]
> best option for Safari users? to stay on Safari 12.1.2 😬

Safari 13 will come with macOS Catalina  later this year, so may not be a good permanent solution for most people.

**Comment by GB-CG** [comment]
Which one from the App Store is the most efficient in terms of CPU/memory/energy?

**Comment by lx4r** [comment]
I'm curious what the differences between Safari and Firefox Beta/Developer/Nightly are in terms of CPU load and battery consumption. Does anyone have a benchmark on that?

**Comment by ghost** [comment]
> Which one from the App Store is the most efficient in terms of CPU/memory/energy?

Do you need whitelisting or constant filter updating? If not, I think they all *should* be the same.

**Comment by ghost** [comment]
> I'm curious what the differences between Safari and Firefox Beta/Developer/Nightly are in terms of CPU load and battery consumption. Does anyone have a benchmark on that?

Haven't seen anything comparing to Safari, but here it is comapred to Chrome.

https://bugzilla.mozilla.org/show_bug.cgi?id=1429522#c45

**Comment by GB-CG** [comment]
> Do you need whitelisting or constant filter updating? If not, I think they all _should_ be the same.

I don't mind. If they theoretically perform the same and does its job then fair enough.
But still, a comparison of all the apps would be nice to see.

**Comment by autigers1970** [comment]
Well my two cents:  Adguard is working well for me combined with the built in Safari content blocker.  I'd been using Unicorn Blocker but something changed with Safari recently and I couldn't fine tune the rules to block things without breaking other things.  Not noticing any added overhead on my 2013 MacBook Pro 13".  The only thing I'll say is that when you add a custom filter in Adguard, it does take 2-3 seconds for it to process before letting you add another.  So if they could improve that performance, that would be great.

**Comment by ameshkov** [comment]
@autigers1970 we definitely will improve the overall UI performance in the next update (v1.6). It's just we were more focused on the adblocking quality, but it seems it's time to focus on other things for some time.

**Comment by nidlol** [comment]
> > best option for Safari users? to stay on Safari 12.1.2 😬
> 
> Safari 13 will come with macOS Catalina later this year, so may not be a good permanent solution for most people.

this is in anyway not a recommendable solution, since there may be severe bugfixes in future releases... was more of a joke... 

Safari is dead. I have to accept it.

**Comment by joz3l** [comment]
Sadly after researching these recommendations, they all feel like consolation prizes compared to uBlock_0. I really hope someone with the same comprehension and standards recreates the project as a compliant(?) Safari extension even with reduced features due to new limitations.

**Comment by alvarnell** [comment]
Sorry, but I don’t see that it would be any better than those already covered here.

Sent from my iPad

-Al-

**Comment by ghost** [comment]
> Sadly after researching these recommendations, they all feel like consolation prizes compared to uBlock_0. I really hope someone with the same comprehension and standards recreates the project as a compliant(?) Safari extension even with reduced features due to new limitations.

I think that AdGuard for Safari does everything that uBO does that can be possibly implemented into Safari.

**Comment by eddyg** [comment]
Using Firefox when [this scrolling issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1124108) is _still_ around? Ick... *Really* going to miss uBO-Safari. 😢

**Comment by roop** [comment]
> It will not possible for uBlock Origin to work with the upcoming Safari 13 / macOS Catalina release

@p4t44 Have you looked at the possibility of creating a [Safari App Extension](https://developer.apple.com/documentation/safariservices/safari_app_extensions) (which is not the same as a content-blocker extension)? You can inject JavaScript into any page, listen to `beforeload` and block those events if required. I've tried these in macOS Catalina and all of this works.

The downside is that there's no "global" area for scripts or resources -- that is now the Swift code, and there's no way to call Swift code and get back an answer synchronously (i.e. no `canLoad` equivalent exists). This implies that all JavaScript needs to get injected into every page and on every reload, which might affect performance and/or memory footprint.

**Comment by ghost** [comment]
> > It will not possible for uBlock Origin to work with the upcoming Safari 13 / macOS Catalina release
> 
> @p4t44 Have you looked at the possibility of creating a [Safari App Extension](https://developer.apple.com/documentation/safariservices/safari_app_extensions) (which is not the same as a content-blocker extension)? You can inject JavaScript into any page, listen to `beforeload` and block those events if required. I've tried these in macOS Catalina and all of this works.
> 
> The downside is that there's no "global" area for scripts or resources -- that is now the Swift code, and there's no way to call Swift code and get back an answer synchronously (i.e. no `canLoad` equivalent exists). This implies that all JavaScript needs to get injected into every page and on every reload, which might affect performance and/or memory footprint.

I'm not the develoeper and don't really understand how any of it works. I just spent far to long scraping together why uBO stopped working in Safari when I upgraded to the Catalina beta, and thought I would try to explain after a lot of posts on /r/ublockorigin had the same issue as me. The developer who made ublock-safari also hasnt updated it for over a year even before Safari 13. But if you or anyone else thinks they can make it work, please do.

**Comment by roop** [comment]
@p4t44 I understand your position now. Sorry, I'm afraid I don't think I can work on this as of now. I'd be happy to help anyone else who wants to take it up, though.

**Comment by frodo2007** [comment]
> ... Though it is still curently possible to install uBlock Origin [by downloading the extension from Github](https://github.com/el1t/uBlock-Safari/releases/download/1.16.0/uBlock0.safariextz) (edit: must follow [these instructions](http://osxdaily.com/2019/02/08/bypass-safari-no-longer-supports-unsafe-extension-error-mac/), it will not be starting from Safari 13 / macOS Catalina, when the legacy entension API will be fully deprecated.

This doesn't work anymore as the “Show Extension Builder” command in the “Develop” menu no longer exists (Safari 13 on macOS 10.14.6).

**Comment by ghost** [comment]
> > ... Though it is still curently possible to install uBlock Origin [by downloading the extension from Github](https://github.com/el1t/uBlock-Safari/releases/download/1.16.0/uBlock0.safariextz) (edit: must follow [these instructions](http://osxdaily.com/2019/02/08/bypass-safari-no-longer-supports-unsafe-extension-error-mac/), it will not be starting from Safari 13 / macOS Catalina, when the legacy entension API will be fully deprecated.
> 
> This doesn't work anymore as the “Show Extension Builder” command in the “Develop” menu no longer exists (Safari 13 on macOS 10.14.6).

That was written before Safari 13 was released, fixed now.

**Comment by eddyg** [comment]
FWIW, if you still haven't updated to Safari 13, you can run:

    softwareupdate --ignore Safari13.0.1MojaveAuto

to hide the Safari 13 update...

**Comment by liam-k** [comment]
Thank you! I’d like to see https://better.fyi added to the list. Costs 2,99 and is open source.

**Comment by ghost** [comment]
> Thank you! I’d like to see https://better.fyi added to the list. Costs 2,99 and is open source.

I disagree that better is a good choice in comparison to the other options. Though I have not personally tested it, it only seems to have one content blocker so its blocking power will be less than alternatives such as AdGuard (free) or Wipr (also $1.99). It seems like an unexceptional paid content blocker from what I have seen about it.

**Comment by asadyryn** [comment]
@p4t44 thank you very much for the advice!!! 1Blocker the best.

**Comment by grahamannett** [comment]
I just posted this: https://www.reddit.com/r/uBlockOrigin/comments/extv6g/would_anyone_be_interested_in_starting_a/?

I haven't found any of the adblockers here as good (dont know much about Safari extensions but AdGuard is an electron app with a safari extension, seems like that might be one of the reasons many of these are not very good?) and wanted to see if this would be of interest to anyone.  Perhaps @el1t would be interested in this as he was the previous maintainer and might be enticed if users are willing to pay up front?  Or even just his input at all about why this would or wouldn't be feasible.

**Comment by alvarnell** [comment]
I think the current developer and those who had hoped to carry on have made it clear that this isn’t a matter of money. Current Apple rules won’t allow anything like uBlock to perform as it used to.

**Comment by grahamannett** [comment]
@alvarnell yeah I deleted the post since I wanted to look into it more.  I found https://github.com/dgraham/Ka-Block which is a good prototype of ideas but am trying to understand more so how ublock works so i can understand if there is a similar approach.  
Im all ears for what apple rules make this not feasible compared to ublock origin (like specific workflows or functions that handle stuff I could see in ublock or where a feature like NSExtensionRequestHandling fails). 

If theres somewhere its spelled out by an actual dev that would be helpful.  From my completely uninformed view it seems like the issue is that all the handling done in js would be done in swift, what parts of the old code can even be reused? I have no idea

**Comment by ghost** [comment]
> @alvarnell yeah I deleted the post since I wanted to look into it more. I found https://github.com/dgraham/Ka-Block which is a good prototype of ideas but am trying to understand more so how ublock works so i can understand if there is a similar approach.
> Im all ears for what apple rules make this not feasible compared to ublock origin (like specific workflows or functions that handle stuff I could see in ublock or where a feature like NSExtensionRequestHandling fails).
> 
> If theres somewhere its spelled out by an actual dev that would be helpful. From my completely uninformed view it seems like the issue is that all the handling done in js would be done in swift, what parts of the old code can even be reused? I have no idea

Take a look at AdGuard for Safari, https://github.com/AdguardTeam/AdGuardForSafari. It is open source and supports as much of the functionality of uBlock Origin as is reasonably feasible in Safari.

As I understand currently an app / extension can send Safari content blocking rules which Safari enforces. This is contary to how uBlock Origin works, where you give the rules to uBO and it enforces them. If you want to create a Safari app / extension which replicated the functionality of uBO in Safari as much as possible that would be great but I'd recommend not using the uBO name / "brand" for the project (as it would not share any of the code or inner workings with uBO, it should go by a different name to avoid confusion. uSafari?). You might also like to use AdGuard as a base.

**Comment by vassudanagunta** [comment]
@grahamannett I don't mean to be mean, but as a New Yorker I can't help but be direct. It seems like you didn't give much weight to all of the effort that went into the explanation at the top of this issue, effort by @p4t44 and people who contributed edits to that explanation via comments.

If you want confirmation from @el1t himself, then take a look at which issue he pinned to [the top of the issues list](https://github.com/el1t/uBlock-Safari/issues). This pin, as far as I know, is his latest and final comment on uBlock for Safari. In addition, the explanation above contains a link to a statement by @gorhill himself.

If you want more technical details, follow the links, e.g. the Hacker News discussion.

**Comment by grahamannett** [comment]
@vassudanagunta its chill. I dont think you are being too mean and yeah I've read this topic many times since it's been posted trying to figure out what the latest news was in terms of other options and to see if others had any actual plans.  

Not trying to be mean either, but did you link me to the pinned issues being this exact issue unintentionally or was that just a way to be snarky without actually providing anything of value? It'd be more helpful and less abrasive if you were to actually say something that could clarify to me why the uBlock origin type network filtering is not feasible in Safari anymore and explain more than just saying "read others posts" since the info is lacking imo.    

I read the HN topic at the time and theres some barebones info about stuff but I haven't read it extensively since I can't be bothered to read 400+ posts of HN users giving bad advice, opining about how terrible the tech company they specifically dont work for is, etc.  

If you have a specific post or blog writeup elsewhere that gives a more technical explanation of any of this, I'd find that incredibly helpful though.  I searched a bit and came up with nothing but tech news sites saying the same thing in 30 different ways.

I still have yet to really understand how this: https://github.com/gorhill/uBlock/wiki/Overview-of-uBlock's-network-filtering-engine works as a whole and why new safari extensions can't mimic at least a large part of uBlock and use bits of the JS stuff for some things.   Also in part because the Apple docs aren't great compared to the mozilla docs for extensions it's unclear where something like NSExtensionRequestHandling overlaps with what parts of uBlock network filtering.  


@p4t44 Yep, I definitely realize now that the name shouldn't use uBlock and I don't want to infringe upon the uBlock name unless someone more official wants to work on it.
  
Yea I spent a fair amount of time looking at the source for AdGuard Safari yesterday and its what I use on safari since it was one of the first recommended on places like this many months ago and the others I've tried are worthless.  I've found it super slow and it has terrible performance issues if you are using the same amount of lists as you would on uBlock.  Using it as a base seems like something that I would have no idea where to begin given the electron app part and then stuff like: https://github.com/AdguardTeam/AdGuardForSafari/blob/master/Shared/AESharedResources.m
(I'm sure it works and is good, but I do not have the desire to dive into that and probably not smart enough).  

Again, I understand that the js extension api stuff isn't going to work as it does with uBlock but uBlock is so good that I just want something even half as good/quick.

**Comment by exwyezed** [comment]
Hey guys, did you check out the new macOS update? could it help us somehow?

![Screen Shot 2020-03-26 at 12 59 25 PM](https://user-images.githubusercontent.com/9363272/77644779-e30cd000-6f61-11ea-977b-f26ec3ce282d.png)

URL: https://developer.apple.com/documentation/macos_release_notes/macos_catalina_10_15_4_release_notes

**Comment by vassudanagunta** [comment]
@alfonmga All Catalyst does is allow UIKit apps (those built for iOS) to be easily ported to run on MacOS. This does not in any way loosen restrictions on Safari plugins/extensions.

Apple has promised its users to build an impenetrable wall around user's privacy. There simply is not a way to allow third party code to do content-filtering without giving that code access to a user's entire browsing history. We can't have our cake and eat it too.

Content Blocker is Apple's solution to this problem: Third parties supply Safari with filtering rules, not code. Unless uBlock Origin adopts that (it won't), uBlock Origin will never work in Safari.

**Comment by ghost** [comment]
@vassudanagunta 

> Apple has promised its users to build an impenetrable wall around user's privacy. There simply is not a way to allow third party code to do content-filtering without giving that code access to a user's entire browsing history. We can't have our cake and eat it too.


Except that just about every content blocker has another extension included which can see your entire browsing history for whitelisiting. They haven't built an impenetrable wall around users privacy, they have built an impenetrable wall around user customisation and choice.

Also you can have your cake and eat it too in any other browser by being careful and not installing malicious extensions.

**Comment by vassudanagunta** [comment]
@p4t44 I think your take is an oversimplification. The beauty of it is they don't have a monopoly; the people who prefer the kind of user customization you want can have it: Firefox. Or Linux. 

But for people who don't want it at the expense of security, even super technical people like me, and people who like me have spent countless hours digging non-technical people out of malware holes, or get pissed off at all the con-artists taking advantage of people confused by complexity (most people, including us technical people), we are glad that we have the choice to choose a more locked down platform. 

Just as you said, it's all about choice and Apple is the only vendor in this space offering that choice.

**Comment by gingerbeardman** [comment]
Another point regarding 1Blocker, it now requires 10.15 Catalina.

Thankfully I'm still using an old version that works on 10.14 Mojave.

**Comment by Baccount** [comment]
Ad-Guard works well on macOS 10.15, it’s open source and you can disable 2 out of its 8 Safari extensions to prevent it from having any access to your website data, it also allows custom filters. 
App Store version: https://apps.apple.com/us/app/adguard-for-safari/id1440147259?mt=12

GitHub source code: https://github.com/adguardteam/adguardforsafari#

**Comment by superlex** [comment]
@Baccount 
I agree, and AdGuard for Safari supports most of uBlock0 syntax (e.g. scriptlets) :)

**Comment by KDederichs** [comment]
Seems like they announced web extensions on the WWDC, could this eventually mean that add-ons like uBlock might be possible again with the next OS version?

**Comment by freediverx** [comment]
I was thinking the same thing when I watched the keynote. I’m a little skeptical, though, because the announcement was light on details, and it wasn’t entirely clear whether something like uBlock Origin can work in Big Sur, or whether there will be functional limitations restricting features like hiding HTML screen elements… Apple is likely to put up barriers on any functionality that could be used as a malware attack vector.

**Comment by KDederichs** [comment]
They do say `New support for the WebExtensions API ` so I'm hopeful that they actually support the whole range of it, with the restriction of the user having to enable the add-on explicitly

**Comment by freediverx** [comment]
I heard that and made a mental note to look it up.  Was that API developed by Mozilla, and was it used to create the uBlock Origin add-on for Firefox? Also, assuming their mention of this related specifically to macOS and not to iOS or iPadOS as well… (I need to rewatch the keynote.)

**Comment by ameshkov** [comment]
More info here:
https://developer.apple.com/documentation/safariservices/safari_web_extensions

Compatibility and limitations:
https://developer.apple.com/documentation/safariservices/safari_web_extensions/assessing_your_safari_web_extension_s_browser_compatibility

The most important limitation for blockers is this part:

* webRequest
    * BlockingResponse not supported.
    * Blocking requests not supported.

It basically means that we have to continue using their Safari Content Blocking API.

**Comment by freediverx** [comment]
Ugh, their Safari Content Blocking API is worthless, at least based on every content blocker I’ve tested. So much for that idea. Damnit, Apple.

**Comment by joshdick** [comment]
> Ugh, their Safari Content Blocking API is worthless, at least based on every content blocker I’ve tested. So much for that idea. Damnit, Apple.

Send feedback to Apple via the Feedback app once you’re able to run a beta! 🙂

**Comment by aaronsky** [comment]
I could be wrong, but those docs look the same as the ones before today. Meaning I'm not so sure they've updated their documentation site with the new APIs

**Comment by ghost** [comment]
https://www.reddit.com/r/uBlockOrigin/comments/hdz0bo/will_ublock_origin_back_to_macos_big_sur/fvoc7wk/

> Apparently not, this is what [it says](https://developer.apple.com/documentation/safariservices/safari_web_extensions/assessing_your_safari_web_extension_s_browser_compatibility) for the `webRequest` API:
> 
> > BlockingResponse not supported.
> > 
> > Blocking requests not supported.
> 
> **Edit:** There are other unsupported API methods:
> 
> - `onCreatedNavigationTarget` (to deal with unwanted popups);
> - minimal support for `insertCSS`: no user styles, can't inject on a per-frame basis;
> - No `privacy` API;

**Comment by speckz** [comment]
FYI, https://hacks.mozilla.org/2020/06/welcoming-safari-to-the-webextensions-community/

**Comment by freediverx** [comment]
Can someone with the requisite expertise look here and see if the APIs required for uBlock Origin features are supported?

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs <https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs>

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/886325/85471450-c445c600-b5b0-11ea-9ea4-efa710d0f4f6.png)

**Comment by amarendra** [comment]
> Send feedback to Apple via the Feedback app once you’re able to run a beta! 🙂

Apple doesn't make what users want, Apple makes what they want users to want. Besides the overwhelming majority of Apple user/fan base is always happy with what Apple makes or tries/claims to make.

**Comment by vegai** [comment]
Here's a fine option if you're comfortable with blocking via /etc/hosts: https://github.com/StevenBlack/hosts.git -- as a bonus, it can optionally block fakenews, porn and other such junk.

**Comment by gorhill** [comment]
@ameshkov Please correct the disinformation in this picture which purports to compare AdGuard for Safari to other blockers on Safari: https://cdn.adguard.com/public/Adguard/Blog/safari-compare.jpg I have no doubt you are well aware that "UBlock" on Safari is not uBO.

**Comment by ameshkov** [comment]
@gorhill uh, wrong logo? sorry for that, we'll fix it. May I ask where you found that picture?

**Comment by gorhill** [comment]
<https://adguard.com/en/adguard-safari/overview.html>, _"What's the difference between AdGuard for Safari and other ad blockers? "_.

**Comment by elesto** [comment]
So.. is ubo coming to new safari and big sur? I dont really understand all this technical speak

**Comment by liamengland1** [comment]
> So.. is ubo coming to new safari and big sur? I dont really understand all this technical speak

No. https://www.reddit.com/r/uBlockOrigin/comments/hdz0bo/will_ublock_origin_back_to_macos_big_sur/fvoc7wk/

Use uBo in firefox or chrome, or use adguard in safari.

**Comment by ghost** [comment]
FYI, it seems that Wipr no longer supports blocking YouTube ads: https://www.reddit.com/r/MacOS/comments/hs16jl/what_is_going_on_with_wipr/

**Comment by chaithanya008** [comment]
> Just FIY I would not recommend buying 1Blocker. I paid both for the initial version and the "1Blocker X" one on IOS but I had the constant feeling the filter lists weren't being updated and I wasn't sure of their origin (I'm assuming they use EasyList for the ads, but which one do they use for privacy & annoyances?).
> 
> AdGuard for Safari is free, open-source and is compatible with uBlock-style filter lists (it has code that does its best to convert them to Safari content blocking lists). You can choose to use EasyList, EasyPrivacy, Fanboy's lists or any of the regional lists so I would recommend it over 1Blocker.

Mac preferences> safari extensions - says permission to read sensitive data including passwords and logins! is that ok?

**Comment by chaithanya008** [comment]
> > So.. is ubo coming to new safari and big sur? I dont really understand all this technical speak
> 
> No. https://www.reddit.com/r/uBlockOrigin/comments/hdz0bo/will_ublock_origin_back_to_macos_big_sur/fvoc7wk/
> 
> Use uBo in firefox or chrome, or use adguard in safari.

safari extension preferences say - permission to read sensitive information including passwords and logins. Do they store these information?

**Comment by alvarnell** [comment]
> Mac preferences> safari extensions - says permission to read sensitive data including passwords and logins! is that ok?

Only if you want it to work, otherwise it won't.

> permission to read sensitive information including passwords and logins. Do they store these information?

Only locally and only what is necessary for its use.

**Comment by jorge-d** [comment]
Hi,
I decided to switch back to Safari as my main browser for battery reason and discovered that uBlock Origin is not compatible anymore. After reading most of the possible solutions I think I came up with a pretty acceptable outcome: using NextDNS.

[Nextdns](https://nextdns.io) is a (free for basic usage) DNS resolver, when you create an account you can select blocklists, which does include for example Easylist:
<img width="891" alt="Screenshot 2020-09-01 at 14 21 09" src="https://user-images.githubusercontent.com/1178099/91856583-a9519b00-ec5e-11ea-8ba3-304cca095e33.png">

The only downside is that there is no menu action on webpage to say "whitelist this website", it has to be done on the configuration page of nextdns.

Cheers,

ps: I'm not affiliated with NextDNS in anyway, just a happy user

**Comment by freediverx** [comment]
NextDNS (and similar services) are a valid way to block unwanted ads and tracking. However, they lack the ability to manually select and block annoying screen elements on the fly. I’m currently using AdGuard as my replacement for uBlock Origin. It’s nowhere near as light or fast, but it restores that ability to hide unwanted annoyances on specific web pages as desired.

**Comment by KDederichs** [comment]
Also should be mentioned that with services like NextDNS you basically reroute all of your net activity through their service which is... let's say not ideal in a privacy kind of sense. (Even if they claim to do nothing with that data)

**Comment by translit** [comment]
> ability to hide unwanted annoyances on specific web pages as desired

Yes, this is key.

**Comment by gorhill** [comment]
> they lack the ability to manually select and block annoying screen elements on the fly

Solutions like NextDNS are also not able to filter network requests through URL pattern-based matching. Large swathes of EasyList and other such lists are URL pattern-based filters. What [I explained here](https://www.reddit.com/r/uBlockOrigin/comments/epjlpw/can_anybody_explain_to_me_why_ublockorigin_works/fejujjn/) also applies to DNS-based blocking.

**Comment by jorge-d** [comment]
I'm not denying all those point, which are completely valid. I'm just saying that for a completely simple remove-ads scenario this is probably the easiest solution, the killer point for me being that those same filters can be applied to mobile trafic.

**Comment by ghost** [comment]
I am going to LOVE MacOS Big Sur since it means either uBO will work again or I can, with a little tweaking, install the FireFox version of uBO on Safari

**Comment by ghost** [comment]
Oh wait...

> > > So.. is ubo coming to new safari and big sur? I dont really understand all this technical speak
> > 
> > 
> > No. https://www.reddit.com/r/uBlockOrigin/comments/hdz0bo/will_ublock_origin_back_to_macos_big_sur/fvoc7wk/
> > Use uBo in firefox or chrome, or use adguard in safari.
> 
> safari extension preferences say - permission to read sensitive information including passwords and logins. Do they store these information?

**Comment by alvarnell** [comment]
All blocking extensions must have permission to read everything on a web page, which may include sensitive information, in order to do it's job. There is no reason for it to retain or do anything else with that data in order to perform it's blocking functions, but one would have to perform a detailed analysis of it's storage and off-platform communications in order to answer your question. I would also encourage you to review it's Privacy Policy for a detailed explanation of what it collects and what it does with same. Then you have to make up your mind on whether you believe what it says and if it's worth any risk you perceive.

Sent from my iPad

-Al-

**Comment by cristian-rivera** [comment]
According to https://arstechnica.com/gadgets/2020/11/macos-11-0-big-sur-the-ars-technica-review/7/:

> Safari 14 gives extension developers a lifeline by allowing them to use the WebExtensions API that allows Chrome and Firefox extensions (and Edge, and Opera, and most browsers based on Google’s or Mozilla’s codebases) to use most of the same code. Using Xcode, you can make Safari Web Extensions by importing your existing Chrome or Firefox extension using a command-line tool provided by Apple.

If true, hoping this will finally bring uBlock back to Safari 🙏🏼

**Comment by ghost** [comment]
@cristian-rivera 

> webRequest
> 
>     BlockingResponse not supported.
> 
>     Blocking requests not supported.

https://developer.apple.com/documentation/safariservices/safari_web_extensions/assessing_your_safari_web_extension_s_browser_compatibility

And also watch this: https://www.youtube.com/watch?v=_dduavvDj6s

**Comment by glmanhtu** [comment]
@el1t is there any chance to bring back this extension ???

**Comment by alvarnell** [comment]
@el1t has not posted here since May of 2017.

**Comment by possiblerobot** [comment]
I want to second the rec for NextDNS. It's fantastic. I use it in combination with a Safari extension called Wipr, which is a clean, no-frills ad and tracker blocker based on Easylist. It has no options and you can't subscribe to arbitrary lists. Even though it has no options, it seems to cover my pattern-based filter needs. I stopped using AdGuard because it's so heavy and sometimes hits the CPU enough to ramp the fans up. That's too much for a blocker. NextDNS catches anything Wipr misses, which is very little. Plus, NextDNS filters stuff outside of Safari, catching random trackers that are built into apps, etc.

Also, here's a pro-tip I just discovered for using NextDNS on MacOS Big Sur: use the NextDNS Apple config file generator. It uses the new native encrypted DNS support and plays nice with my VPN.

https://apple.nextdns.io/

**Comment by possiblerobot** [comment]
Just to be clear, if an *efficient*, full-featured blocker like uBlock existed for Safari, I'd use it. I just got tired of having an Electron behemoth hogging resources in the background ::looks at AdGuard::. Right now, Wipr is using 20mb RAM total, and it's been running for a couple days.

@freediverx and @translit are right. We need pattern-based blocking inside webpages in addition to DNS-level blocking. And we need to be able to subscribe to any blocklist we'd like.

**Comment by superlex** [comment]
@possiblerobot 
1) Open AG for Safari
2) Update your filters
3) Close AG for Safari

No app in background needed, no resource used :)

**Comment by possiblerobot** [comment]
@superlex That's not a bad strategy, for sure. But computers are supposed to take care of boring/annoying background tasks for us. Opening AG once or twice a day to update lists is exactly what we should *not* have to do. But maybe in a week, that's what I'll be doing too. Haha.

At least Safari 14 took a step in the right direction with standard (mostly) WebExtensions. I still have hope.

**Comment by superlex** [comment]
@possiblerobot 
Yep, you're right.. We need a simple script in order to update automatically filters.

**Comment by dwiehoff** [comment]
> Switch to a different browser.

From what I can tell, only Safari supports using **AirPlay** with embedded videos. Is there a way to stream videos via AirPlay within Chrome or Firefox?

**Comment by muHashh** [comment]
So what's the recommended one for M1 macs?

**Comment by elesto** [comment]
Honestly I gave up trying to mess with safari and I switched to firefox and ubo. Been happy ever since.

**Comment by cristian-rivera** [comment]
> Honestly I gave up trying to mess with safari and I switched to firefox and ubo. Been happy ever since.

Agreed, the ability to customize, low resource usage and overall security/privacy settings made Firefox a no-brainer for me.

**Comment by praveenperera** [comment]
> > Honestly I gave up trying to mess with safari and I switched to firefox and ubo. Been happy ever since.
> 
> Agreed, the ability to customize, low resource usage and overall security/privacy settings made Firefox a no-brainer for me.

Unfortunately Safari is still king when it comes to battery usage

**Comment by fdenzer** [comment]
If

- the websites don't work
(Safari is the new IE9 to web devs like myself: hard to test against and so much of niche, that a "install a different, mainstream browser" nag screen is getting my preferred solution. But of course, some users don't even know what a browser is and on mobile devices you can not change away from Apples Rendering Engine.)

- and/or I am forced to watch ads,

then I do lose more battery, than by using FF+μBO.

**Comment by BOTKooper** [comment]
> If
> 
> * the websites don't work
>   (Safari is the new IE9 to web devs like myself: hard to test against and so much of niche, that a "install a different, mainstream browser" nag screen is getting my preferred solution. But of course, some users don't even know what a browser is and on mobile devices you can not change away from Apples Rendering Engine.)
> * and/or I am forced to watch ads,
> 
> then I do lose more battery, than by using FF+μBO.

Well, it works just fine, and if you wanna show "install a different browser" to 50% mobile internet, god bless you do not.
Rendering is actually a point, sure. Yeah, it's strange sometimes, but what else we could get from so closed (from communication point) company. Nothing restricts you from committing to webkit though.

For me the only features missing there is multi-profile like in Chrome for differing personal and work accounts and support for ordinary extensions like uBlock, but for that I have Chrome installed.

**Comment by vprelovac** [comment]
Just as a heads up, [Orion](https://browser.kagi.com) (macOS browser, WebKit fork) has native support for uBlock Origin and many other web extensions. We are currently in private beta and accepting new testers.

**Comment by vassudanagunta** [comment]
@vprelovac

> Just as a heads up, [Orion](https://browser.kagi.com) (macOS browser, WebKit fork) has native support for uBlock Origin and many other web extensions. We are currently in private beta and accepting new testers.

I put it on HN for you: https://news.ycombinator.com/item?id=26631598

**Comment by strafe** [comment]
@vprelovac will you be accepting some applications soonish?

**Comment by vprelovac** [comment]
Current waitlist time for Orion is about 2-3 weeks.

**Comment by ujeshkrishnan** [comment]
The Safari support for WebExtensions was released in 2021 and is available from Safari 14 (shipped with macOS Big Sur). WebExtensions for Safari are published in the App Store. [See documentation here](https://developer.apple.com/documentation/safariservices/safari_web_extensions). 

Any hope for uBlock back in Safari?

**Comment by ghost** [comment]
@ujeshkrishnan https://www.reddit.com/r/uBlockOrigin/comments/nura67/ubo_on_ios_soon/

**Comment by chinmaythosar** [comment]
Curious if on macOS NetworkExtensions can be used to make a DoT / DoH app (I made one for iOS a while back) which then has lists from ublock (or rather any list) that is added to blacklist for the app such that any DNS request is then blocked. With M1 Macs I feel that searching through anything around 50,000 blacklists won't be much of a problem. Not sure if this approach was discussed anywhere else.

Note : In case anyone is wondering using System DNS in Network Extensions for macOS 11 and iOS 14 will override any DNS from apps. This is because apps can decide to use their own encrypted DNS in iOS 14 / macOS 11 in which case your local DNS blockers will not work.

**Comment by dronezzzko** [comment]
@chinmaythosar there're two options (both worked for me):

- Paste [AdAway hosts list](https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt) into your local hosts file
- Use LittleSnitch and prepared [Little Snitch Rules](https://rulegroups.com/)

**Comment by ink-splatters** [comment]
> AdGuard is SLOOOOOOOOW. Horrible piece of software. Typical of garbage created with Electron.
> […](#)

well, I don't agree it's garbage, it's pretty decent in terms of it does the thing well, but it's painfully slow. And Electron is garbage, of course

**Comment by natejgrimm** [comment]
Safari extensions seem to have new permissions, could uBlock make a comeback to Safari? Rolling with a new version of 1Blocker currently.

**Comment by vassudanagunta** [comment]
Has anyone tried using alternative open source router firmware, such as [Tomato](https://www.polarcloud.com/tomato/) and its forks such as [FreshTomato](https://freshtomato.org), that supports blocking at the router?

From what I can tell, this is achieved via a router-based DNS that "blocks all ads and trash websites via the DNS cache poisoning method". I don't know if other open source router firmware offers other methods.

For Tomato and its forks:
- Using custom scripts to [Set Up Universal Ad Blocking Through Your Router](https://lifehacker.com/set-up-universal-ad-blocking-through-your-router-5060053)
- [You can do a lot of other things with custom scripts](https://wiki.dd-wrt.com/wiki/index.php/Useful_Scripts)

In addition, [FreshTomato has a built-in ad blocker](https://wiki.freshtomato.org/doku.php/adblock) that I guess is easier to manage than custom scripts:
> FreshTomato's ad blocker works through DNS cache poisoning. It downloads lists of URLs/domains to block. It then replaces whatever dnsmasq resolves as the advertiser's correct IP address with an address of 0.0.0.0. 

> FreshTomato's built-in ad blocker is separate from the ad blocking scripts seen on the Web pasted into FreshTomato's custom script window

**Comment by strafe** [comment]
> Has anyone tried using alternative open source router firmware, such as [Tomato](https://www.polarcloud.com/tomato/) and its forks such as [FreshTomato](https://freshtomato.org), that supports blocking at the router?

Yes with OpenWRT. It's not as convenient as blocking at the browser though as exceptions take longer to input and can't be temporarily added with a single click. Best for devices that don't support system-wide blocking, e.g. iOS.

**Comment by Chasapple4** [comment]
Would the Chromium MV3 style work with the new Safari extension model?

**Comment by emikaadeo-git** [comment]
Is there are some plans to make a port of new experimental [uBOLite MV3 API-based content blocker](https://github.com/gorhill/uBlock/releases/tag/uBOLite_0.1.22.9196) to Safari?
Safari have support for MV3 extensions since version 15.4 https://webkit.org/blog/12445/new-webkit-features-in-safari-15-4/#safari-web-extensions
**EDIT:**
Looks like currently it's not possible:
https://github.com/uBlockOrigin/uBOL-issues/issues/52

**Comment by vprelovac** [comment]
Just a heads up that full (or nearly full) uBO experience is possible in a WebKit-based browser. Orion browser, a WebKit-based Mac browser (still in beta) supports Chrome/Firefox extensions natively.

https://browser.kagi.com

**Comment by ankitsharma07** [comment]
I have been using Orion and it's been great so far.

**Comment by sdykae** [comment]
My god orion did it 😫

**Comment by cocode32** [comment]
The biggest problem with what everyone has mentioned with using FireFox or a different browser that supports uBlock Origin is that on iOS and iPadOS; those browsers don't support extensions that are available for those browsers, because Apple has forced them to wrap their browser in webkit so that it is basically a skinned version of Safari.

The OP's message suggestions is the best that I've found. For the past 2 years, I've been looking for a way to get rid of all the tracking and ads and such on my iPhone but I've not had any success; until now - when I found this post that detailed everything.

I decided to bite the bullet at take the subscription to AdGuard (bought the lifetime license). Because of what the OP mentioned about AdGuard using the same syntax as uBlock Origin, all I had to do was migrate my filters into my AdGuard on my phone and now everything is working as it should; balance has been restored 😁

From all the reading that I've done the past week, AdGuard is my number 1 recommendation, but all the suggestions that the OP mentioned in the original message are good choices too 😊

**Comment by F1248** [comment]
> > permission to read sensitive information including passwords and logins. Do they store these information?
> 
> Only locally and only what is necessary for its use.

Has anyone ever verified this in the source code?
This app could intercept ALL passwords, bank details, etc., as well as scan the entire browsing history.
I know that the permission to read web content is necessary to block certain items. But why does it also need to have access to the history?
I would be very grateful if a professional could check this independently!

**Comment by alvarnell** [comment]
@F1248 Your comment appears to be way off-topic for this very old issue which has largely been abandoned by anybody that matters.

The quotes you used are not even found in this discussion, so you might have better luck getting help with your request if you started a new Issue.

**Comment by daslicht** [comment]
any news on this?

**Comment by alvarnell** [comment]
There has been no real news on this topic since the developer's original posting over four years ago, so I don't anticipate there will ever be.

**Comment by daslicht** [comment]
> There has been no real news on this topic since the developer's original posting over four years ago, so I don't anticipate there will ever be.

thats sad, I really like safari, but without proper adblockewr it is kind of mehhh

**Comment by marlonjames71** [comment]
2024 and I'm wondering why uBlock O can't be ported to Safari still. Didn't Apple make improvements in this area?

**Comment by sdykae** [comment]
Plugin extension api is not fully supported in Safari, UBlock O uses this api for its core functionality, It can't be ported cuz that

> 2024 and I'm wondering why uBlock O can't be ported to Safari still. Didn't Apple make improvements in this area?

**Comment by daslicht** [comment]
> Plugin extension api is not fully supported in Safari, UBlock O uses this api for its core functionality, It can't be ported cuz that
> 
> > 2024 and I'm wondering why uBlock O can't be ported to Safari still. Didn't Apple make improvements in this area?

How does Adguard manage it to make it work ?

**Comment by vprelovac** [comment]
Adguard uses content blocking functionality in WebKit which is less powerful than native requests-based web extension APIs used by uBO. 

If you want to see full uBO functionality in a WebKit browser, [Orion browser](https://kagi.com/orion) has implemented web extensions API on top of WebKit and natively supports Chrome and Firefox extensions. This support is still in beta but most popular extensions work (including uBO). Disclaimer: I am one of the devs of Orion.

**Comment by daslicht** [comment]
> Adguard uses content blocking functionality in WebKit which is less powerful than native requests-based web extension APIs used by uBO.
> 
> If you want to see full uBO functionality in a WebKit browser, [Orion browser](https://kagi.com/orion) has implemented web extensions API on top of WebKit and natively supports Chrome and Firefox extensions. This support is still in beta but most popular extensions work (including uBO). Disclaimer: I am one of the devs of Orion.

I dont care which engine :) I just want to use privacy gateway of safari and its convenient macos integration. Or does Orion can be used with apples privacy gateway?

[EDIT] I juist installed orion and a excited, it just looks like Safari anbd even support touch id etc, keep testing

**Comment by daslicht** [comment]
> Orion.

does Orion iOS also support extensions now in EU?

**Comment by marlonjames71** [comment]
> Adguard uses content blocking functionality in WebKit which is less powerful than native requests-based web extension APIs used by uBO.
> 
> If you want to see full uBO functionality in a WebKit browser, [Orion browser](https://kagi.com/orion) has implemented web extensions API on top of WebKit and natively supports Chrome and Firefox extensions. This support is still in beta but most popular extensions work (including uBO). Disclaimer: I am one of the devs of Orion.

I must admit that I haven't been using Orion recently due to some bugs I encountered and a preference for a more polished design similar to what I found in Beam browser that was around a little bit ago. However, I'm optimistic about the improvements Orion is making, especially with the implementation of web extensions API. I look forward to giving it another try in the future. It's got so many great features and I especially appreciate how privacy focused it is.

**Comment by gitguys** [comment]
> 2024 and I'm wondering why uBlock O can't be ported to Safari still. Didn't Apple make improvements in this area?

IMHO, I wonder if it's worth it?  I've switched all my Macs to Firefox a long time ago and don't regret it.  Firefox is fast nowadays and I've dodged numerous security and privacy vulnerabilities over time because Safari was the gateway for multi-chain macOS attacks.  I used to prefer Safari, but I just gave Firefox a Safari-looking theme and went forward.  Far less headaches, far more options and often far ahead of Safari in features built-in and third party.  I don't know if I'll ever go back to Safari on macOS unless something very compelling is offered I can't do with Firefox.

**Comment by daslicht** [comment]
> > 2024 and I'm wondering why uBlock O can't be ported to Safari still. Didn't Apple make improvements in this area?
> 
> IMHO, I wonder if it's worth it? I've switched all my Macs to Firefox a long time ago and don't regret it. Firefox is fast nowadays and I've dodged numerous security and privacy vulnerabilities over time because Safari was the gateway for multi-chain macOS attacks. I used to prefer Safari, but I just gave Firefox a Safari-looking theme and went forward. Far less headaches, far more options and often far ahead of Safari in features built-in and third party. I don't know if I'll ever go back to Safari on macOS unless something very compelling is offered I can't do with Firefox.

Same here except that touchID, keychain, passkey works better in Safari, which theme do you use ?

**Comment by maxexcloo** [comment]
> IMHO, I wonder if it's worth it? I've switched all my Macs to Firefox a long time ago and don't regret it. Firefox is fast nowadays and I've dodged numerous security and privacy vulnerabilities over time because Safari was the gateway for multi-chain macOS attacks. I used to prefer Safari, but I just gave Firefox a Safari-looking theme and went forward. Far less headaches, far more options and often far ahead of Safari in features built-in and third party. I don't know if I'll ever go back to Safari on macOS unless something very compelling is offered I can't do with Firefox.

It is a shame to miss some features like 2FA auto fill though!

**Comment by gitguys** [comment]
> which theme do you use ?

Now that I look closer I'm just using a theme called "System theme - auto" and it looks similar to Safari to me.  I think in the past I used to use a different theme to simulate Safari but forgot I switched maybe to the default after it improved?

There's this following option perhaps, but I'm happy with "System theme — auto":

https://github.com/AdamXweb/WhiteSurFirefoxThemeMacOS

I haven't had keychain issues with Firefox, but I use the 1Password extension for password management for the most part.  I haven't noticed any touchID issues with Firefox either, but maybe that's due to using 1Password, I don't know.

**Comment by daslicht** [comment]
Another issue is that Appel Privacy Gateway and 'hide my email' is only working in Safari afaik
Or does that work in other browsers as well?

**Comment by gitguys** [comment]
@daslicht I'm not sure, I don't use "hide my email" but I do know Firefox has its own free "Firefox Relay" extension.

https://nerdschalk.com/firefox-relay-vs-apple-hide-my-email-what-to-use-and-why/

As far as Apple Private Gateway, I use my own VPN that works with every connection on my Mac computers and isn't specific to any app.  Although I do have the option for split-tunneling to allow certain apps through the VPN but I don't have any need for that.  I would hope that APG would protect any app that connects like any standard VPN does, but I don't use it.  It may not play well with Firefox because Apple is notorious for trying to edge out third party options for their userbase.  Another reason why I use my own VPN, Firefox, etc.

**Comment by daslicht** [comment]
> @daslicht I'm not sure, I don't use "hide my email" but I do know Firefox has its own free "Firefox Relay" extension.
> 
> https://nerdschalk.com/firefox-relay-vs-apple-hide-my-email-what-to-use-and-why/
> 
> As far as Apple Private Gateway, I use my own VPN that works with every connection on my Mac computers and isn't specific to any app. Although I do have the option for split-tunneling to allow certain apps through the VPN but I don't have any need for that. I would hope that APG would protect any app that connects like any standard VPN does, but I don't use it. It may not play well with Firefox because Apple is notorious for trying to edge out third party options for their userbase. Another reason why I use my own VPN, Firefox, etc.

Firefox relay sounds great but paying extra for a dedicated VPN is out of sc ope over here.

**Comment by gitguys** [comment]
> Firefox relay sounds great ...

One caveat (unless Mozilla has changed this recently) is it apparently has a ridiculously small attachment size limit according to the article (150 KB), so it may not be the best choice either (unless Moz upgraded that limit since the article was published).

**Comment by HVR88** [comment]
>Plugin extension api is not fully supported in Safari, UBlock O uses this api for its core functionality, It can't be ported cuz that

No one has mentioned what's missing from CURRENT versions of Safari. Apple claim that's there's a green light to support extension conversion from other browsers. Some people balk at developer registration being $99 - except you can get a free account and distribute outside the app store if you so choose.

**Comment by callmefil** [comment]
Does anyone know if Hyperweb extension for Safari works at all with UBlock Origin script being uploaded? Or is there more to it? Can someone test it out for me?

**Comment by jr-seven** [comment]
[](url)

**Comment by affan2021shaikh** [comment]
Add wBlock to Top picks
